google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/AppEngineDeploymentRunConfigurationEditor.java

```diff
@@ -246,26 +253,48 @@ protected JComponent createEditor() {
   /**
    * A somewhat generic way of generating a file for a {@link TextFieldWithBrowseButton}.
    */
-  private static class GenerateConfigActionListener implements ActionListener {
+  private class GenerateConfigActionListener implements ActionListener {
 
     private final Project project;
     private final String fileName;
     private final TextFieldWithBrowseButton filePicker;
     private final Supplier<File> sourceFileProvider;
+    private final JPanel fileSelector;
 
     public GenerateConfigActionListener(
         Project project,
         String fileName,
         Supplier<File> sourceFileProvider,
-        TextFieldWithBrowseButton filePicker) {
+        TextFieldWithBrowseButton filePicker,
+        JPanel fileSelector) {
       this.project = project;
       this.fileName = fileName;
       this.sourceFileProvider = sourceFileProvider;
       this.filePicker = filePicker;
+      this.fileSelector = fileSelector;
     }
 
     @Override
     public void actionPerformed(ActionEvent event) {
+      if (sourceFileProvider.get() == null) {
+        if (!isUserSpecifiedPathDeploymentSource()) {
+          throw new AssertionError("By our design, artifacts should always have .jar or .war; "
+            + " sourceFileProvider will return a correct config file.");
```

Comment:
Oh since its an exception message not a comment, how about something like this:

"Error generating configuration file: artifact deployment source is missing a source .jar or .war file."